tests/npm-app/src/ccf/builtin.ts

```diff
@@ -0,0 +1,212 @@
+     */
+    query: string;
+
+    /**
+     * An object with ``text()``/``json()``/``arrayBuffer()`` functions
+     * to access the request body in various ways.
+     */
+    body: Body<T>;
+
+    /**
+     * An object describing the authenticated identity retrieved
+     * by this endpoint's authentication policies.
+     *
+     * ``caller.policy`` is a string indicating which policy accepted this request,
+     * for use when multiple policies are listed.
+     * The other fields depend on which policy accepted;
+     * most set ``caller.id``, ``caller.data``, and ``caller.cert``,
+     * while the ``"jwt"`` policy sets ``caller.jwt``.
+     */
+    caller: any;
+  }
+
+  export type ResponseBodyType<T> = string | ArrayBuffer | JsonCompatible<T>;
+
+  export interface Response<T extends ResponseBodyType<T> = any> {
+    /**
+     * (Optional) The HTTP status code to return.
+     * Defaults to ``200``, or ``500`` if an exception is raised.
+     */
+    statusCode?: number;
+
+    /**
+     * (Optional) An object mapping lower-case HTTP header names to their values.
+     * The type of ``body`` determines the default value of the ``content-type`` header.
+     */
+    headers?: { [key: string]: string };
+
+    /**
+     * (Optional) The body of the response.
+     * Either
+     * a `string <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String>`_ (``text/plain``),
+     * an `ArrayBuffer <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer>`_ (``application/octet-stream``),
+     * a `TypedArray <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray>`_ (``application/octet-stream``),
+     * or as fall-back any `JSON-serializable <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify>`_ value (``application/json``).
+     *
+     * The content type in parentheses is the default and can be overridden in ``headers``.
+     */
+    body?: T;
+  }
+
+  export type EndpointFn<
+    A extends JsonCompatible<A> = any,
+    B extends ResponseBodyType<B> = any
+  > = (request: Request<A>) => Response<B>;
+
+  /**
+   * A map in the Key Value Store.
+   *
+   * ``KVMap`` is modelled after JavaScript's ``Map`` object,
+   * except that keys and values must be of type ``ArrayBuffer``
+   * and no guarantees on iteration order are provided.
+   */
+  export interface KVMap {
+    has(key: ArrayBuffer): boolean;
+    get(key: ArrayBuffer): ArrayBuffer | undefined;
+    set(key: ArrayBuffer, value: ArrayBuffer): KVMap;
+    delete(key: ArrayBuffer): boolean;
+
+    /**
+     * @param callback A function with parameters ``value, key, kvmap``.
+     */
+    forEach(
+      callback: (value: ArrayBuffer, key: ArrayBuffer, kvmap: KVMap) => void
+    ): void;
+  }
+
+  export type KVMaps = { [key: string]: KVMap };
+
+  export interface WrapAlgoParams {
+    name: string;
+  }
+
+  /**
+   * TODO
```

Comment:
@wintersteiger This is for you :)

Reply:
@wintersteiger I moved the sources to src/js. Do you want to fill in the jsdoc in this PR or in a separate one? I don't mind either way.

Reply:
Let's do it in a separate one, easier to review.

Reply:
https://github.com/microsoft/CCF/issues/2317